### PR TITLE
IAC 745 Implementation of Various Types of vRLI Fallback Objects

### PR DESCRIPTION
**Pull request by akantchev**
…ts\n Signed-off-by: Alexander Kantchev

### Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added relevant error handling and logging messages to help troubleshooting
- [x] I have added necessary documentation (if appropriate)
- [x] I have updated CHANGELOG.md with a short summary of the changes introduced
- [x] I have tested against live environment, if applicable
- [x] Dependencies in pom.xml are up-to-date
- [x] My changes have been rebased and squashed to the minimal number of relevant commits
- [x] My changes have a descriptive commit message with a short title, including a `Fixed #XXX -` or `Closed #XXX -` prefix to auto-close the issue

### CODE REVIEW

**Review by Michaelpalacce — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 ## v2.29.1 - 15 Feb 2023
 
 ### Fixes
+* [maven-plugins-vrli] IAC-745 / VRLI Alerts Fallback Object Set to LogInsight Only During Push
```
[the remark this answers is not in the document]
Comment (akantchev) [reply CHANGELOG.md:11]: Actually should be [ maven-plugin-vrli ] similar to  [maven-plugins-vrli] IAC-741 as it is an extension of it.

[the remark this answers is not in the document]
Comment (Michaelpalacce) [reply CHANGELOG.md:11]: The idea in the changelog is to specify (ideally) the maven module where the changes were done in. This makes it artifact-manager. Not sure why [maven-plugins-vrli] was decided for 741.

File: common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/RestClientVrliV2.java [hunk 2]
```diff
@@ -157,34 +158,30 @@ private void rewriteVcopsIntegrationInfo(AlertDTO alert) {
...
                     String.format("Unable to update vCOPs integration for alert '%s', unable to fetch vROPs resources: %s", alert.getName(), e.getMessage()));
         }
-        if (resourceDto == null) {
-            return;
-        }
-
-        Optional<ResourcesDTO.ResourceList> vrliResource = resourceDto.getResourceList().stream()
-                .filter(item -> item.getResourceKey().getResourceKindKey().equalsIgnoreCase(VRLI_RESOURCE_KEY_TYPE)).findFirst();
-        if (!vrliResource.isPresent()) {
+        String targetResourceName = alert.getRecipients().getVrops().getVcopsResourceName();
```
[the remark this answers is not in the document]
Comment (akantchev) [reply common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/RestClientVrliV2.java:165]: Unfortunately this cannot be unit tested, as it requires a special type of alerts that have vROPs integration, it has been tested against a live vRLI environment.

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@
 ## v2.29.1 - 15 Feb 2023
 
 ### Fixes
+* [artifact-manager] IAC-745 / VRLI Alerts Fallback Object Set to LogInsight Only During Push
```
Comment (n-vodenicharski) [inline CHANGELOG.md:14]: this is in the wrong place. This section is for previous release

**Review by n-vodenicharski — DISMISSED** [review]

**Review by ivo-kotev — APPROVED** [review]

**Pull request merged**